src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -23,3 +28,67 @@ export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
 
   return Uint8Array.from(colors.flat());
 }
+
+export function usePanZoom(camera: OrthographicCamera): ReactThreeFiber.Events {
+  const [startOffsetPosition, setStartOffsetPosition] = useState<Vector3>();
+
+  const onPointerDown = useCallback(
+    (evt: PointerEvent) => {
+      evt.stopPropagation();
+
+      const { currentTarget, pointerId } = evt as React.PointerEvent;
+      currentTarget.setPointerCapture(pointerId);
+
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+      setStartOffsetPosition(camera.position.clone().add(projectedPoint));
```

Comment:
When dragging starts, I save a **start offset position**, which is the sum of the positions of the camera and of the mouse pointer (projected from world to camera space). When the pointer moves during the drag, I compute the camera's new position by subtracting the pointer's projected position from the start offset position.